### DISCUSSION TITLE
Uvicorn not reloading on file update + installed Watchfiles

### DISCUSSION THREAD

**Discussion by starzar** [Kludex/uvicorn#1973] [Potential Issue]
Uvicorn  is not reloading on file update(.csv,.html etc  except .py ) .It does specify that it will watch the provided directory but on file update there is no reload. 
However  the standalone Watchfiles package is independently able to detect same file updates. 

 Watchfiles package has been installed separately. 

Any additional settings for reloading on file update?
```
INFO:     Will watch for changes in these directories: ['H:\\NseScraping\\pandasPivot\\webApp\\htmlPdTables']
INFO:     Uvicorn running on http://127.0.0.1:80 (Press CTRL+C to quit)
INFO:     Started reloader process [7964] using WatchFiles
INFO:     Started server process [21276]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:63536 - "GET /sandata/ HTTP/1.1" 200 OK
INFO:     127.0.0.1:63536 - "GET /favicon.ico HTTP/1.1" 404 Not Found
WARNING:  WatchFiles detected changes in 'main.py'. Reloading...
INFO:     Shutting down
INFO:     Waiting for application shutdown.
INFO:     Application shutdown complete.
INFO:     Finished server process [21276]
INFO:     Started server process [9428]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
```

**Comment by kddove85** [comment]
Hello,

Assuming that you are using Windows and have a Docker instance, I encountered a similar issue while working on a Django application that utilizes vue. To resolve this, I added the following command to my run command:

```
--reload-include *.vue
```
This enabled uvicorn to detect changes to my .vue files. However, there is a larger issue that needs to be addressed. Although uvicorn recognized the changes, the application displayed a message indicating that it had restarted. Nevertheless, when I refreshed the page, the changes were not reflected.

The root cause of the problem may be attributed to uvicorn's continued usage of watchgod instead of watchfiles. https://pypi.org/project/watchgod/.

**Comment by robertoandrade** [comment]
Seeing the same behavior here running uvicorn on Python for Windows. Any suggestions for correction?
